yjit/src/codegen.rs

```diff
@@ -2539,6 +2539,24 @@ fn gen_concatstrings(
     KeepCompiling
 }
 
+// Return Some if whether stack-top objects are fixnums or not at least at compile
+// time is known, otherwise None.
+fn two_fixnums_on_stack(jit: &mut JITState, ctx: &mut Context) -> Option<bool> {
+    if jit_at_current_insn(jit) {
+        let comptime_recv = jit_peek_at_stack(jit, ctx, 1);
+        let comptime_arg = jit_peek_at_stack(jit, ctx, 0);
+        return Some(comptime_recv.fixnum_p() && comptime_arg.fixnum_p());
+    }
+
+    let recv_type = ctx.get_opnd_type(StackOpnd(1));
+    let arg_type = ctx.get_opnd_type(StackOpnd(0));
+    match (recv_type, arg_type) {
+        (Type::Fixnum, Type::Fixnum) => Some(true),
+        (Type::Unknown | Type::UnknownImm, Type::Unknown | Type::UnknownImm) => None,
+        _ => Some(false),
+    }
+}
```

Comment:
Seems like a good idea. The only thing I would change is, for consistency, maybe `two_fixnums_on_stack` should be a method of the Context in `core.rs`. It follows a similar pattern to other context methods that use context info to return `Some(known_type)` or `None`. I think that if it followed this pattern and was used as `ctx.two_fixnums_on_stack()` it would make it easier to understand what it does by reading the code.

Reply:
:+1: moved the method to Context https://github.com/ruby/ruby/pull/7168/commits/6bdf8f9fbe2e671e756e69affcecd43e66c3b6b0